CONTRIBUTING.md

```diff
@@ -6,11 +6,77 @@ _As contributors and maintainers of this project, and in the interest of fosteri
 
 ## Getting Started
 
-We have full documentation on how to get started contributing here:
+### Repo layout
+
+```
+├── publishers # Home for all functions metadata
+│   ├── kustomize
+│   │   ├── fn-foo.yaml
+│   │   ├── fn-bar.yaml
+│   │   ├── README.md
+│   │   └── OWNERS # OWNERS of the publisher
+│   ├── kubeflow
+│   ├── sig-cli
+│   └── OWNERS # OWNERS to approve new publishers
+├── krm-functions # Home for in-tree functions source code
+│   ├── Makefile
+│   ├── kustomize
+│   │   ├── fn-foo
+│   │   ├── README.md
+│   │   └── OWNERS # OWNERS to approve code change to the function
+│   └── sig-cli
+├── site 
+├── Makefile
+└── OWNERS
+```
+
+## Contributing in-tree KRM function source code
```

Comment:
We should talk about the criteria to accept functions as in-tree functions. For example, we should not accept 3rd party functions as in-tree functions, since if they give up the maintenance, it will be a problem for us.

Reply:
What do we consider 3rd party functions? If some company wishes to donate a function, does it need to be under SIG-CLI? In that case, would there be any publishers in the `krm-functions` folder outside of `krm-functions/sig-cli`?

Reply:
I added a requirements section to `krm-functions/README.md` stating that we cannot accept 3rd party functions as in-tree functions